### PR TITLE
change elsif condition with switch case.

### PR DESCRIPTION
**Pull request by seuros**

### CODE REVIEW

**Comment by olleolleolle** [comment]
v1.4.0 has been released, and the circular require issue is no longer there. Can this be closed?

**Comment by seuros** [comment]
@olleolleolle i replaced the `ifs` condition with a `switch case`

File: lib/gyoku/xml_value.rb [hunk 1]
```diff
@@ -16,22 +16,25 @@ class << self
...
-        elsif Date === object
+        when Date
           object.strftime XS_DATE_FORMAT
-        elsif String === object
+        when String
           escape_xml ? CGI.escapeHTML(object) : object
-        elsif object.respond_to?(:to_datetime)
-          create object.to_datetime
-        elsif object.respond_to?(:call)
+        when Proc
```
Comment (olleolleolle) [inline lib/gyoku/xml_value.rb:28]: Q: I wonder if my custom `#call`-supporting instance would match this `when`, or if the `object.respond_to?(:call)` needs to go down  into the `else`.

Comment (seuros) [reply lib/gyoku/xml_value.rb:28]: the instance is not a Proc, so it we need to add it to the else block.

Comment (olleolleolle) [reply lib/gyoku/xml_value.rb]: Thanks for your care ❤️!

**Pull request merged**